tests/plugins/plugin.url.test.js

```diff
@@ -120,4 +120,13 @@ describe('Plugin: URL controller', () => {
     expect(window.history.replaceState).toHaveBeenCalledTimes(0);
   })
 
+  test('only q= param is selected from url query params', () => {
+    const INTIAL_URL="http://127.0.0.1:8080/BookReaderDemo/demo-internetarchive.html?ocaid=adventuresofoli00dick&q=foo"
+    br.options.urlMode = 'hash';
+    br.init();
+    const result = br.urlParamsFiltersOnlySearch(INTIAL_URL);
+    const expectedRes = new URLSearchParams({ q: "foo" }).toString();
+    expect(result).toBe(expectedRes);
```

Comment:
In general, unit tests should be kind of "dumb"; just using flat data. Otherwise you end up with sort of a tautology. Do this without `URLSearchParams`, and have another test to handle a url with a special character in it's `q` (e.g. `?q=foo%20boo`)

Reply:
Actually, the function is essentially a static function (since it doesn't use `this`, so you can call it as `BookReader.prototype.urlParamsFiltersOnlySearch('https://...')` without having to initialize BR in any way.